include/mra/kernels/multiply.h

```diff
@@ -11,64 +11,126 @@ namespace mra {
   template<mra::Dimension NDIM>
   SCOPE size_type multiply_tmp_size(size_type K) {
     const size_type K2NDIM = std::pow(K,NDIM);
-    return 4*K2NDIM; // workspace, r, r1, and r2
+    return 25*K2NDIM; // workspace, r1*8, cnodeA*8, cnodeB*8,
   }
 
   namespace detail {
 
     template <typename T, Dimension NDIM>
     DEVSCOPE void multiply_kernel_impl(
       const Domain<NDIM>& D,
+      const Key<NDIM>& keyA,
+      const Key<NDIM>& keyB,
       const TensorView<T, NDIM>& nodeA,
       const TensorView<T, NDIM>& nodeB,
+      TensorView<T, NDIM+1>& cnodeA,
```

Comment:
Why do you call them `cnode`? For compressed? Aren't we working on function coefficients here?

Reply:
`cnodesA`<=> child nodes of A